data/src/main/java/com/example/data/datasource/remote/ReqResPagingSource.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.data.datasource.remote
+
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import com.example.data.entity.Follower
+import com.example.data.service.MainService
+import retrofit2.HttpException
+import java.io.IOException
+import javax.inject.Inject
+
+class ReqResPagingSource @Inject constructor(
+    private val mainService: MainService
+) : PagingSource<Int, Follower>() {
+    override fun getRefreshKey(state: PagingState<Int, Follower>): Int? {
+        return state.anchorPosition?.let { anchorPosition ->
+            state.closestPageToPosition(anchorPosition)?.prevKey?.plus(1)
+                ?: state.closestPageToPosition(anchorPosition)?.nextKey?.minus(1)
+        }
+    }
+
+    override suspend fun load(params: LoadParams<Int>): LoadResult<Int, Follower> {
+        val nextPageNumber = params.key ?: STARTING_PAGE_INDEX
+        return try {
+            val response = mainService.getFollowers(nextPageNumber)
+            LoadResult.Page(
+                data = response.data,
+                prevKey = null,
```

Comment:
잉 근ㄴ데 이제 확인했는데, prevKey상태가 왜이러져 ㅋㅋㅋㅋ 이러면 prevKey가 업데이트 안될것같은데 아래처럼 고치면 좋을 것 같네여~

```suggestion
        prevKey = if (key == 1) null else key - 1,
```

Reply:
설명이 좀 부족해서 덧붙이자면

```
val key = params.key ?: 1
val response = mainService.getFollowers(nextPageNumber)
      LoadResult.Page(
                data = response,
                prevKey = if (key == 1) null else key - 1,
                nextKey = if (response.isEmpty()) null else key + 1
            )
        }
```

이런식으로 관리하라는 말입니다..ㅋㅋㅋ

Reply:
수정했습니다 ~ ! ㅋㅋㅋㅋ